### ISSUE TITLE
Changing `tabSize` to `0` causes freezing

### ISSUE THREAD

**Issue by shshaw** [codemirror/CodeMirror#6301]
## Steps to replicate:

1. Go to an example, like https://codemirror.net/demo/visibletabs.html
2. In the browser console, run `editor.setOption('tabSize', 0)`. Visually, the 0 tab size displays correctly.
3. Go to an indented line and hit Enter
4. CodeMirror instance will freeze, and the Browser may even freeze

I suspect there's some logic when creating a new line to try to get the proper indent level that's failing because `tabSize` is `0`.

This doesn't seem to be a problem when creating a CodeMirror instance with `tabSize` set to `0`, but happens when switching after it's created.

Definitely occurs in Chrome 80+.

**Comment by shshaw** [comment]
This is a scenario from a user of CodePen.

I don't understand the desire for zero-width tabs, but it is a request we've had 🤷

**Comment by marijnh** [comment]
I don't think it makes sense to support this. Would it help to default to 1 when a value <1 is given?
